src/vllm_haystack/__init__.py

```diff
@@ -101,7 +101,12 @@ def __init__(
             if key in kwargs
         }
 
-        self.tokenizer = Tokenizer.from_pretrained(tokenizer or model_name_or_path,auth_token=hf_token)
+        if isinstance(tokenizer, str) or tokenizer is None:
+            self.tokenizer = Tokenizer.from_pretrained(tokenizer or model_name_or_path,auth_token=hf_token)
+        else:
+            if not hasattr(tokenizer, "encode"):
+                raise AttributeError(f"tokenizer `{type(tokenizer)}` does not provide an `encode` method")
```

Comment:
We also have to ensure a `decode` function exists, as we will have to decode the tokens if we need to truncate a prompt.

Reply:
Ah yes. Good call!